server/src/handlers/auth_handler.rs

```diff
@@ -631,3 +672,92 @@ pub async fn login_cli() -> Result<HttpResponse, ServiceError> {
     })?;
     Ok(HttpResponse::Ok().content_type("text/html").body(html_page))
 }
+
+#[derive(Deserialize, Debug, ToSchema)]
+pub struct CreateApiUserBody {
+    pub user_email: String,
+    pub user_name: String,
+}
+
+#[derive(Serialize, Debug, ToSchema)]
+pub struct CreateApiUserResponse {
+    pub user: SlimUser,
+    pub organization_id: uuid::Uuid,
+    pub api_key: String,
+}
+
+#[utoipa::path(
+    post,
+    path = "/auth/create_api_only_user",
+    context_path = "/api",
+    tag = "Auth",
+    responses(
+        (status = 200, description = "The user id", body = CreateApiUserResponse),
+        (status = 400, description = "Error message indicitating you are not currently signed in", body = ErrorResponseBody),
+    ),
+    security(
+        ("ApiKey" = ["readonly"]),
+    )
+)]
+pub async fn create_api_only_user(
+    req: HttpRequest,
+    body: web::Json<CreateApiUserBody>,
+    pool: web::Data<Pool>,
+) -> Result<HttpResponse, ServiceError> {
+    let access_key = req
+        .headers()
+        .get("X-Shopify-Authorization")
+        .ok_or(ServiceError::BadRequest(
+            "X-Shopify-Authorization header is required".to_string(),
+        ))?
+        .to_str()
+        .map_err(|_| {
+            ServiceError::BadRequest("Failed to parse X-Shopify-Authorization header".to_string())
+        })?;
+
+    if access_key != get_env!("SHOPIFY_SECRET_KEY", "SHOPIFY_SECRET_KEY must be set") {
+        return Err(ServiceError::BadRequest(
+            "Invalid session token".to_string(),
+        ));
+    }
```

Comment:
You could implement this as middleware, but it's fine to wait and do it later the next time we are doing something that needs this shopify header.